org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/view/task/SelectionDependentHandler.java

```diff
@@ -28,29 +32,30 @@
 
     @Override
     public void setEnabled(Object evaluationContext) {
-        boolean enabled;
-        TaskView taskView = getTaskView();
-        if (taskView != null) {
-            NodeSelection selectionHistory = taskView.getSelection();
-            enabled = isEnabledFor(selectionHistory);
-        } else {
-            enabled = false;
+        if (evaluationContext instanceof IEvaluationContext) {
+            IEvaluationContext evalContext = (IEvaluationContext) evaluationContext;
+            // getting the current ISelection from the IEvaluationContext
+            // See https://wiki.eclipse.org/Command_Core_Expressions
+            Object variable = evalContext.getVariable(ISources.ACTIVE_CURRENT_SELECTION_NAME);
+            if (variable instanceof ISelection) {
+                setBaseEnabled(isEnabledFor(NodeSelection.from((IStructuredSelection) variable)));
+            }
```

Comment:
Use `if (variable instanceof NodeSelection)` statement and remove the `NodeSelection.from()` invocation.